utils/test/context/context.util.tsx

```diff
@@ -4,46 +4,75 @@
  */
 import React, { Provider, ReactElement, ReactNode } from 'react';
 import { render, RenderResult } from '@testing-library/react';
+import {
+  _ConfigFeatureFlag_TestProvider,
+  defaultClientConfig,
+  apolloQueryResponseType,
+  defaultConfigFeatureFlagValue,
+} from 'Contexts';
 
-type ProviderEntry = {
-  value: unknown;
-  provider: Provider<unknown>;
+type TestProviderWithValue<T> = {
+  value: T;
+  provider: Provider<T>;
 };
 
-const contextWrapper = (
-  providers: Array<ProviderEntry>,
+const contextWrapper: <T>(
+  providers: Array<TestProviderWithValue<T>>,
   children: ReactNode
-) => {
-  const provider = providers.shift();
-  if (provider) {
-    if (providers.length == 0) {
-      return (
-        <provider.provider value={provider.value}>{children}</provider.provider>
-      );
-    } else {
-      return (
-        <provider.provider value={provider.value}>
-          {contextWrapper(providers, children)}
-        </provider.provider>
-      );
-    }
-  } else {
-    throw new Error(
-      'Provider array is empty. Please provide at least one provider'
-    );
-  }
-};
+) => JSX.Element = (providers = [], children = null) => (
+  <React.Fragment>
+    {providers.reduceRight(
+      (accJsx, { provider: ProviderToMount, value }) => (
+        <ProviderToMount value={value}>{accJsx}</ProviderToMount>
+      ),
+      children
+    )}
+  </React.Fragment>
+);
 
-const renderWithContextProviders = (
+const renderWithContextProviders: <T>(
   ui: ReactElement,
   options: Record<string, unknown>,
-  providers: Array<ProviderEntry>
-): RenderResult =>
+  providers: Array<TestProviderWithValue<T>>
+) => RenderResult = (ui, options, providers) =>
   render(ui, {
     wrapper: ({ children }) => {
       return contextWrapper(providers, children);
     },
     ...options,
   });
 
-export { renderWithContextProviders };
+/** withConfigFeatureFlagContext renders the given `ui` JSX with a ConfigFeatureFlag provider seeded with default values. Use this if your components indirectly use/require a value from the ConfigFeatureFlag context, else use `withCustomConfigFeatureFlagContext` so the value from ConfigFeatureFlag can be controlled  */
+const withConfigFeatureFlagContext: (
```

Comment:
Given the naming of these functions, I'd assume that they would be used like:

```
render(withConfigFeatureFlagContext(...))
```

But the render function isn't needed. Could these be modified so they work in that way or renamed to `renderWithCustomConfigFeatureFlagContext` so that the names are consistent with the behaviour

Reply:
Agreed - will be updated in the next commit